flang/include/flang/Lower/FIRBuilder.h

```diff
@@ -251,6 +253,11 @@ class FirOpBuilder : public mlir::OpBuilder,
     return getRegion().getParentOfType<mlir::FuncOp>();
   }
 
+  const fir::KindMapping &getKindMap() { return kindMap; }
+
+  mlir::Value convertOnAssign(mlir::Location loc, mlir::Type toTy,
```

Comment:
What is the benefit of the location argument compared to using the one of the `FIROpBuilder` ? The interface of the builder helpers are starting to diverge here, half have it, the other does not.

Reply:
I'm a little worried about caching the location in the Builder object. It's not clear that it is always being set before use and then there is clearly a problem with reentrancy. I think there is even some shadow builders being created to hide that. I'm starting to lean towards having the bridge keep the location information and pass it to the builder instead.

Reply:
OK, I see your point.

Reply:
Does this have implications for `create` vs. `createHere`?  I was wondering if I should be using `createHere` in more cases.